app/javascript/packages/verify-flow/index.tsx

```diff
@@ -41,6 +42,11 @@ export function VerifyFlow({ initialValues = {}, basePath, appName }: VerifyFlow
         promptOnNavigate={false}
         basePath={basePath}
         titleFormat={`%{step} - ${appName}`}
+        onStepChange={(step) => {
+          if (step.name === 'personal_key_confirm') {
+            trackEvent('IdV: personal key submitted');
```

Comment:
@aduth  I think this is the part I'm most unsure about.. it is intended to track when a personal key is submitted but currently it is triggered when the confirm modal is opened instead

Reply:
I also wasn't sure the best way to go about testing this one.. should I create a spec for VerifyFlow component?

Reply:
> @aduth I think this is the part I'm most unsure about.. it is intended to track when a personal key is submitted but currently it is triggered when the confirm modal is opened instead

`onStepChange` would be called with the _new_ step, not the step that was submitted. We could consider passing both the new and previous steps as arguments. We might want to do that anyways to avoid having the "visited" events in the `FormSteps` implementation.

I might suggest looking at this hook for reference, which holds a previous value, and might be useful for what we're trying to do:

https://github.com/18F/identity-idp/blob/a013cede5d029ea4edb38a92223cca1ae19280c1/app/javascript/packages/document-capture/hooks/use-previous.js

e.g.

```tsx
const previousStep = usePrevious(step);
useDidUpdateEffect(() => onStepChange(step, previousStep));
```

One thing this doesn't account for is the "visited" event for the very first step. Maybe we just need to handle that from `VerifyFlow` ?

```tsx
useEffect(() => {
  trackEvent(stepToVisitedEvent(STEPS[0].name));
}, []);
```

Reply:
> I also wasn't sure the best way to go about testing this one.. should I create a spec for VerifyFlow component?

That sounds like a good idea. We probably also want updated specs for `FormSteps` passing the step argument through `onStepChange`.

I started implementing a `VerifyFlow` spec in #6229 if you want to pull some of that in here for reference.

Reply:
>We could consider passing both the new and previous steps as arguments. We might want to do that anyways to avoid having the "visited" events in the FormSteps implementation.

On second thought, this could get tricky for handling things like history, or returning to an earlier step. We wouldn't want to say that the previous step was "submitted" in those cases. Maybe we need a new prop callback like `onStepSubmit`.